components/dss-tools/plugins/org.wso2.integrationstudio.artifact.dataservice/src/org/wso2/integrationstudio/artifact/dataservice/ui/wizard/GenerateDataServicesWizardPage.java

```diff
@@ -0,0 +1,1133 @@
+     * failed.
+     */
+    private void updatePageStatus() {
+        String singleServiceName = dataModel.getServiceName();
+        File dataService = new File(dataModel.getLocation().getAbsolutePath() + File.separator + 
+                "dataservice" + File.separator + singleServiceName + ".dbs");
+        if (avalableDatasources == null || avalableDatasources.size() <= 1) {
+            updatePageStatus(DataServicesWizardConstants.ErrorMessages.DATASOURCE_NOT_FOUND);
+        } else if (selectedDatasource == null) {
+            updatePageStatus(DataServicesWizardConstants.ErrorMessages.DATASOURCE_NOT_SELECTED);
+        } else if (allAvailableTables == null || allAvailableTables.size() <= 0) {
+            updatePageStatus(DataServicesWizardConstants.ErrorMessages.TABLE_NOT_FOUND);
+        } else if (selectedTables == null || selectedTables.size() <= 0) {
+            updatePageStatus(DataServicesWizardConstants.ErrorMessages.SELECTED_TABLES_EMPTY);
+        } else if (dataService.exists()) {
+            updatePageStatus("Data service with the name '" + singleServiceName + "' already exists");
+        } else {
+            updatePageStatus(null);
+        }
+    }
+    
+
+    /**
+     * Update the page state.
+     *
+     * @param msg
+     *            error message
+     */
+    private void updatePageStatus(String msg) {
+        setErrorMessage(msg);
+         setPageComplete(msg == null);
+    }
+    
+    private void updateTablesMap(Button selectedButton) {
+        if (selectedButton.getSelection()) {
+            String [] cell = buttonMap.get(selectedButton);
+            String eventTableName = cell[0];
+            String eventMethod = cell[1];
+            EnumSet<Methods> currentMethods = selectedTables.get(eventTableName);
+            currentMethods.add(Methods.valueOf(eventMethod));
+            selectedTables.put(eventTableName, currentMethods);
+        } else {
+            String [] cell = buttonMap.get(selectedButton);
+            String eventTableName = cell[0];
+            String eventMethod = cell[1];
+            EnumSet<Methods> currentMethods = selectedTables.get(eventTableName);
+            currentMethods.remove(Methods.valueOf(eventMethod));
+            selectedTables.put(eventTableName, currentMethods);
+        }
+    }
+    
+    private Map<String, Document> getAvailableDatasources() {
+        String datasourceArtifcatCategory = "datasource/datasource";
+        Map<String, Document> availableDatasourcesMap = new LinkedHashMap<String, Document>();
+        availableDatasourcesMap.put("Select datasource", null);
+        IProject[] projects = ResourcesPlugin.getWorkspace().getRoot().getProjects();    
+        File projectPath = null;
+        for (IProject workspaceProject : projects) {
+            try {
+                if (workspaceProject.hasNature("org.wso2.developerstudio.eclipse.datasource.project.nature")) {
```

Comment:
First check, `workspaceProject.isOpen() && ` and then it has the DS_PROJECT nature logic here? Otherwise, if someone closes the project, this line will throw an error saying the project is closed.

Reply:
Shall we replace the `org.wso2.developerstudio.eclipse.datasource.project.nature` value from `Constants.DS_PROJECT_NATURE` variable?

Reply:
Done https://github.com/wso2/integration-studio/pull/890/commits/7c67b03c80855728fbecedbfaad2e6f19af7f2c6